File: test_go/factorial.go
package main

import (
	"math/big"
)

func _factor(akk *big.Int, n int64) *big.Int {
	if n == 1 {
		return akk
	} else {
		return _factor(big.NewInt(0).Mul(akk, big.NewInt(n)), n-1)
	}
}

func Fac(n int64) *big.Int {
	return _factor(big.NewInt(1), n)
}

File: test_c++/src/factorial/factorial.cpp
#include "factorial/factorial.h"

long long int Factorial::_factorial(long long int akk, long long int n)
{
    if (n == 0)
    {
        return akk;
    }
    else
    {
        return _factorial(akk * n, n - 1);
    }
}

long long int Factorial::factorial(long long int n)
{
    return _factorial(1, n);
}